static/faq.html

```diff
@@ -291,6 +294,55 @@ <h2 id="default-connections">
                 </li>
             </ul>
 
+            <h2 id="securing-data-at-rest">
+                <a href="#securing-data-at-rest">How does GrapheneOS secure data at rest?</a>
+            </h2>
+            <p>GrapheneOS makes full use of the hardware-bound encryption in modern Android devices.
+            The Titan M, a specially designed hardware security module on Pixel 3 handsets handles
+            validation of the user's pincode or password. Upon getting the correct password, the HSM
+            will calculate an access token from the password, from a secret contained in the
+            confines of the HSM, and from a hash of a two-kilobyte partition at the start of the
+            phone's SSD by means of a key derivation function. The HSM will only be able to release
+            the token to the phone so it can in turn derive the keys to decrypt the files if it gets
+            the correct password.</p>
+
+            <p>This system ensures that an adversary in possession of your phone cannot simply
+            extract the SSD and connect it to a much more powerful system to run guessing attacks on
+            it offline, and instead must go through the specific HSM on the phone. The HSM is
+            about the size of a grain of rice and is difficult to attack. Each file is encrypted
+            with AES-256-XTS, and the filesystem metadata is encrypted with AES-256-HEH.</p>
+
+            <p>File Based Encryption on GrapheneOS is always on, enforced, and cannot be switched
+            off or fall back to plaintext. The Titan M hardware security module has its own internal
+            timer which cannot be accessed from the host operating system, and can enforce
+            progressively lengthening delays between guesses at the password. On enough failed
+            guesses, the Titan M can increase the time between guesses to allow only one guess
+            per day. At this rate, to search through all the combinations of a four-digit pin code
+            from 0000 to 9950 would take over 27 years, or for a relatively weak two-word passphrase
```

Comment:
Good thing you brought this up, since after you mentioned it I decided I'd give it a test using my own phone to see if it actually took more than 50 to make it from 30 seconds to a minute and up and up. Originally I was under the impression that it would behave by doubling the time interval, but I decided to experiment a little bit and test with a password, pincode, and pattern, and my results actually differed a bit from what I expected:

After the first five attempts the phone will display the number of incorrect password attempts in a dialogue box, disable the fingerprint sensor and reject all input for 30 seconds. After the next five attempts, the phone will display the number of incorrect password attempts in a dialogue box and reject all input for another 30 seconds. After each unsuccessful attempt at that point, it'll do the same.

I'll actually have to revisit this section, whoops.

Reply:
The Titan M is only involved for the initial unlock after rebooting. Make sure you're properly testing this.

Reply:
I followed the instructions and tested it to 61 results this time. The time delay starts to increment from 30 seconds to 60 seconds at 40 tries, after which the time lockout doubles to two minutes at 50 tries, then four minutes at 60 tries. I think that provided that it continues to work this way, then it would take 100 tries to reach one hour, and then from there after 150 tries it would reach one day between attempts. 

While I was at it, I also plugged a keyboard into my phone and attempted to enter the password, and it looks like GrapheneOS won't allow this which would make automated password spoofing even more difficult. 

Looks like it is working exactly as it is intended to, I just was mistaken and thought that it would be incrementing faster. I'll update the draft with this new information.

Reply:
It allows attaching a keyboard but only detects it at boot or when unlocked, by default.